dspot/src/main/java/fr/inria/diversify/utils/AmplificationHelper.java

```diff
@@ -344,33 +347,55 @@ private static CtMethod cloneMethod(CtMethod method, String suffix) {
      */
     private static CtMethod cloneTestMethod(CtMethod method, String suffix) {
         CtMethod cloned_method = cloneMethod(method, suffix);
-        CtAnnotation testAnnotation = cloned_method.getAnnotations().stream()
-                .filter(annotation -> annotation.toString().contains("Test"))
-                .findFirst().orElse(null);
-        if (testAnnotation != null) {
-            cloned_method.removeAnnotation(testAnnotation);
-        }
         final Factory factory = method.getFactory();
         prepareTestMethod(cloned_method, factory);
         return cloned_method;
     }
 
+    /**
+     * Prepares the test annotation of a method
+     *
+     * @param cloned_method The test method to modify
+     * @param factory The factory to create a new test annotation if needed
+     */
     public static void prepareTestMethod(CtMethod cloned_method, Factory factory) {
-        CtAnnotation testAnnotation;
-        testAnnotation = factory.Core().createAnnotation();
-        CtTypeReference<Object> ref = factory.Core().createTypeReference();
-        ref.setSimpleName("Test");
-
-        CtPackageReference refPackage = factory.Core().createPackageReference();
-        refPackage.setSimpleName("org.junit");
-        ref.setPackage(refPackage);
-        testAnnotation.setAnnotationType(ref);
-
-        Map<String, Object> elementValue = new HashMap<>();
-        elementValue.put("timeout", timeOutInMs);
-        testAnnotation.setElementValues(elementValue);
-
-        cloned_method.addAnnotation(testAnnotation);
+        CtAnnotation testAnnotation = cloned_method.getAnnotations().stream()
+                .filter(annotation -> annotation.toString().contains("Test"))
+                .findFirst().orElse(null);
+        if (testAnnotation != null) {
+            CtExpression originalTimeout = testAnnotation.getValue("timeout");
+            if (originalTimeout == null) {
+                testAnnotation.addValue("timeout", timeOutInMs);
+            } else {
+                int valueOriginalTimeout;
+                if (originalTimeout.toString().endsWith("L")) {
+                    String stringTimeout = originalTimeout.toString();
+                    valueOriginalTimeout = toIntExact(parseLong(stringTimeout.substring(0, stringTimeout.length() - 1)));
+                } else {
+                    valueOriginalTimeout = parseInt(originalTimeout.toString());
+                }
+                if (valueOriginalTimeout < timeOutInMs) {
+                    CtLiteral newTimeout = new CtLiteralImpl<Integer>();
+                    newTimeout.setValue(timeOutInMs);
+                    originalTimeout.replace(newTimeout);
```

Comment:
Please, use the `factory` rather than the constructor:`factory.createCtLiteral(timeOutInMs);`
Also, I would prefer to use `testAnnotation.addValue("timeout, factory.createCtLiteral(timeOutInMs));` instead of replacement of the literal inside the Annotation. In this way, we have some check done by **Spoon** on the values, and possibly capture bugs.

Reply:
About the `addValue`, I tried that but the result was something like `expected = 1000,10000`. And there isn't a `removeValue`.

Reply:
Ah okay, so I guess we will do like that, thank you very much.